buildcheck/backend/yolo_processor.py

```diff
@@ -0,0 +1,167 @@
+    
+    def calculate_bbox_center(self, bbox: list) -> geom.Point:
+        x1, y1, x2, y2 = bbox
+        center_x = (x1 + x2) / 2
+        center_y = (y1 + y2) / 2
+        return geom.Point(center_x, center_y)
+    
+    def create_symbol_from_detection(self, class_name: str, bbox: list) -> Symbol:
+        x1, y1, x2, y2 = bbox
+        
+        # Create BBox object with four corner points
+        bbox_obj = BBox(
+            a=Point(x1, y1),  # Top-left
+            b=Point(x2, y1),  # Top-right
+            c=Point(x2, y2),  # Bottom-right
+            d=Point(x1, y2)   # Bottom-left
+        )
+        
+        # Map class name to category
+        category = self.map_class_to_category(class_name)
+        
+        # Create symbol
+        symbol = Symbol(category, bbox_obj)
+        
+        return symbol
+    
+    def find_room_for_symbol(self, center_point: geom.Point) -> Room:
+        for room in self.layout.rooms:
+            # Create a polygon for each room
+            room_polygon = geom.Polygon([(p.x, p.y) for p in room.junctions])
+            
+            if room_polygon.contains(center_point):
+                return room
+        
+        return None
+    
+    def yoloProcesser(self, confidence_threshold: float = 0.25):
+        # Run YOLO detection and associate symbols with rooms
+        
+        # Run YOLO inference
+        results = self.model.predict(
+            source=self.image_path, 
+            conf=confidence_threshold,
+            save=False,  # Don't save automatically
+            verbose=False
+        )
+        
+        total_detections = 0
+        symbols_assigned = 0
+        
+        # Process each result
+        for result in results:
+            # Get detection data
+            if result.boxes is None or len(result.boxes) == 0:
+                print("No objects detected.")
+                continue
+                
+            boxes = result.boxes.xyxy.cpu().numpy()  # [x1, y1, x2, y2]
+            classes = result.boxes.cls.cpu().numpy()  # Class IDs
+            names = result.names  # Class ID to name mapping
+            confs = result.boxes.conf.cpu().numpy()  # Confidence scores
+                        
+            for i in range(len(boxes)):
+                x1, y1, x2, y2 = map(int, boxes[i])
+                class_id = int(classes[i])
+                confidence = float(confs[i])
+                class_name = names[class_id]
+                
+                total_detections += 1
+                
+                # Create symbol from detection
+                symbol = self.create_symbol_from_detection(
+                    class_name, [x1, y1, x2, y2]
+                )
+                
+                # Calculate center point
+                center = self.calculate_bbox_center([x1, y1, x2, y2])
+                
+                # Find which room contains this symbol
+                containing_room = self.find_room_for_symbol(center)
```

Comment:
This assumes that a symbol is associated with one and only one room. That may or may not be true.

For instance, Dr. said that we probably want to associate a door with both the room and the hallway that it connects to. As mentioned earlier, we should refactor this to return a list of applicable rooms.